dist/docker,docs: replace "--experimental" with "--experimental-features"

tchaikov opened the pull request:
The "--experimental" option was removed in commit https://github.com/tchaikov/scylladb/commit/f6cca741eaa21c4c34362305c240058758ee3ddb. Using this deprecated option now causes Scylla to fail with the error:

```
error: the argument ('on') for option '--experimental-features' is invalid
```

So, in this change, let's update the docker entry point script to use `--experimental-features` command line option instead. The related document is updated accordingly.

Fixes scylladb/scylladb#22207

---

it's an improvement in document mainly targeting developers willing to try out experimental features, hence no need to backport.

scylladb-promoter commented:
## Docs Preview :book:
Docs Preview for this pull request is available [here](http://previews.docs.scylladb.com/scylladb/22283)

**Changed Files:**
- [faq](http://previews.docs.scylladb.com/scylladb/22283/faq)
- [operating-scylla/procedures/tips/best-practices-scylla-on-docker](http://previews.docs.scylladb.com/scylladb/22283/operating-scylla/procedures/tips/best-practices-scylla-on-docker)

**Note:** This preview will be available for 30 days and will be automatically deleted after that period. You can manually trigger a new build by committing changes.

dkropachev commented:
@tchaikov , don't you also need to delete it from docker distrib files:
https://github.com/scylladb/scylladb/blob/514ef48d752fec9c6b421a3452b7a6db8b58c541/dist/docker/scyllasetup.py#L149-L150
and 
https://github.com/scylladb/scylladb/blob/d4f15633691c16d2805fa76bc76f98f6ac2995d4/dist/docker/commandlineparser.py#L7

tchaikov commented:
@dkropachev hi Dmitry, indeed, i didn't realize that our docker came with its own entry point script. so instead of removing the `--experimental` option, i replaced it with `--experimental-features`, so that user can enable individual features using `--experimental-features` option provided by the docker image.

tchaikov commented:
v2:

- updated the docker entry point script to replace `--experimental-features` command line option with `--experimental`.
- updated the commit message and cover letter accordingly.

tchaikov commented:
@dkropachev hi Dmitry, could you please help re-review this change?

dkropachev reviewed (approved):
Looks good, thanks.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14419/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1123/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2762/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/530/)
:white_check_mark: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/35/)
:x: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13891/)
:white_check_mark: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5882/)
:white_check_mark: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/34/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14419/)
#### Failed Tests (1/36922):
- [test_simple_add_new_node_while_schema_changes[True]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14419/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/Tests___dtest___test_simple_add_new_node_while_schema_changes_True_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_simple_add_new_node_while_schema_changes[True])
#### Build Details:
- Duration: 5 hr 17 min
- Builder: spider7.cloudius-systems.com

nyh commented on docs/faq.rst:
```diff
@@ -104,14 +104,35 @@ You can clean snapshots by using :doc:`nodetool clearsnapshot </operating-scylla
+
+   On Docker
+
+   :code:`$ docker exec -it your_node vi /etc/scylla/scylla.yaml`
```
Do we have "vi" installed on our docker? :-)

tchaikov replied:
good question, i am not sure. this command line was inherited from 6e97b83b6028326f00b823c4f8f2da4fa5c50d3c . probably we should revisit it later on. @dkropachev do you happen to known if our docker image has `vi` installed?

nyh commented on docs/operating-scylla/procedures/tips/best-practices-scylla-on-docker.rst:
```diff
@@ -289,17 +289,17 @@ To disable developer mode:
 .. code-block:: console
 
-  docker run --name some-scylla -d scylladb/scylla --experimental 1
+  docker run --name some-scylla -d scylladb/scylla --experimental-feature=udf
```
You are missing an "s" in experimental-feature**s**.

nyh reviewed (changes requested):
Looks good except one typo (missing s) in an example.

tchaikov commented:
v3:

- added the missing "s" in the document in the command line example.

tchaikov commented:
@nyh hi Nadav, thank you for your review. could you please take another look?

nyh commented:
By the way, the dtest failure above is a known flakiness, https://github.com/scylladb/scylladb/issues/22073. I ignored it because documentation (and docker) changes can't cause dtests to fail :-)

tchaikov commented:
> By the way, the dtest failure above is a known flakiness, #22073. I ignored it because documentation (and docker) changes can't cause dtests to fail :-)

thank you for approving this pull request and for identifying the issue which led to the test failure of the previous build.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14432/)
:white_check_mark: - [Docker Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-docker/1124/)
:white_check_mark: - [Rpm Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-centos-rpm/36/)
:x: - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/5890/)
:white_check_mark: - [Offline-installer Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-offline-installer/531/)
:x: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-tablets/2770/)
:white_check_mark: - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/13899/)
:white_check_mark: - [Deb Artifact Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci-unified-deb/35/)
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/14432/)
#### Failed Tests (6/36936):
- [test_change_field_type_with_cdc[Single_cluster]](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14432/testReport/junit/cdc_test/TestCdc/Tests___dtest_with_topology_changes___test_change_field_type_with_cdc_Single_cluster_) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_change_field_type_with_cdc[Single_cluster])
- [test_simple_restart_streaming_node_while_bootstrapping](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14432/testReport/junit/update_cluster_layout_tests/TestUpdateClusterLayout/Tests___dtest_with_tablets___test_simple_restart_streaming_node_while_bootstrapping) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_simple_restart_streaming_node_while_bootstrapping)
- [test_view_build_status_with_replace_node.dev.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14432/testReport/junit/topology_custom/test_view_build_status/Tests___Unit_Tests___test_view_build_status_with_replace_node_dev_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_view_build_status_with_replace_node.dev.2)
- [test_view_build_status_with_replace_node.dev.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14432/testReport/junit/topology_custom/test_view_build_status/Tests___Unit_Tests___test_view_build_status_with_replace_node_dev_2_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_view_build_status_with_replace_node.dev.2)
- [test_tablet_repair_task_list.debug.3](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14432/testReport/junit/topology_tasks/test_tablet_tasks/Tests___Unit_Tests___test_tablet_repair_task_list_debug_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tablet_repair_task_list.debug.3)
- [test_tablet_repair_task_list.debug.3](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/14432/testReport/junit/topology_tasks/test_tablet_tasks/Tests___Unit_Tests___test_tablet_repair_task_list_debug_3_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_tablet_repair_task_list.debug.3)
#### Build Details:
- Duration: 10 hr
- Builder: i-0d238b60a4b43586a (m5ad.8xlarge)

nyh commented:
Wow, what a gallery of flaky tests :-( I'll update all the issues we have on all of them...
